# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/quake/hotfix-x11-notify.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

From: René Rebe <rene@exactco.de>
Subject: x11: wait for MapNotify after mapping the window

Without a window manager no ConfigureNotify follows XMapWindow and the
client waited forever; MapNotify is always sent.

Signed-off-by: René Rebe <rene@exactco.de>

--- a/libs/video/targets/context_x11.c
+++ b/libs/video/targets/context_x11.c
@@ -562,7 +562,7 @@ X11_CreateWindow (int width, int height)
 
 	XMapWindow (x_disp, x_win);
 
-	X11_WaitForEvent (ConfigureNotify);
+	X11_WaitForEvent (MapNotify);
 
 	vid_context_created = true;
 	if (vid_fullscreen->int_val) {
